packages/fiori/src/FlexibleColumnLayout.ts

```diff
@@ -0,0 +1,821 @@
+	* @public
+	*/
+	@slot()
+	endColumn!: Array<HTMLElement>;
+
+	initialRendering: boolean;
+	_handleResize: () => void;
+	static i18nBundle: I18nBundle;
+	_prevLayout: FCLLayout | null;
+
+	constructor() {
+		super();
+
+		this._prevLayout = null;
+		this.initialRendering = true;
+		this._handleResize = this.handleResize.bind(this);
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return FlexibleColumnLayoutCss;
+	}
+
+	static get template() {
+		return FlexibleColumnLayoutTemplate;
+	}
+
+	static get dependencies() {
+		return [Button];
+	}
+
+	static async onDefine() {
+		FlexibleColumnLayout.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	static get ANIMATION_DURATION() {
+		return getAnimationMode() !== AnimationMode.None ? 560 : 0;
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResize.bind(this));
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+	}
+
+	onAfterRendering() {
+		if (this.initialRendering) {
+			this.handleInitialRendering();
+			return;
+		}
+
+		this.syncLayout();
+	}
+
+	handleInitialRendering() {
+		this._prevLayout = this.layout;
+		this.updateLayout();
+		this.initialRendering = false;
+	}
+
+	handleResize() {
+		if (this.initialRendering) {
+			return;
+		}
+
+		// store the previous layout
+		const prevLayoutHash = this.columnLayout!.join();
+
+		// update the column layout, based on the current width
+		this.updateLayout();
+
+		// fire layout-change if the column layout changed
+		if (prevLayoutHash !== this.columnLayout!.join()) {
+			this.fireLayoutChange(false, true);
+		}
+	}
+
+	startArrowClick() {
+		this.arrowClick({ start: true, end: false });
+	}
+
+	endArrowClick() {
+		this.arrowClick({ start: false, end: true });
+	}
+
+	arrowClick(options: { start: boolean, end: boolean }) {
+		// update public property
+		this.layout = this.nextLayout(this.layout, { start: options.start, end: options.end })!;
+
+		// update layout
+		this.updateLayout();
+
+		// fire layout-change
+		this.fireLayoutChange(true, false);
+	}
+
+	updateLayout() {
+		this._width = this.widthDOM;
+		this._columnLayout = this.nextColumnLayout(this.layout);
+		this._visibleColumns = this.calcVisibleColumns(this._columnLayout);
+		this.toggleColumns();
+	}
+
+	syncLayout() {
+		if (this._prevLayout !== this.layout) {
+			this.updateLayout();
+			this._prevLayout = this.layout;
+		}
+	}
+
+	toggleColumns() {
+		this.toggleColumn("start");
+		this.toggleColumn("mid");
+		this.toggleColumn("end");
+	}
+
+	toggleColumn(column: string) {
+		let columnWidth;
+		let columnDOM;
+
+		if (column === "start") {
+			columnWidth = this.startColumnWidth;
+			columnDOM = this.startColumnDOM;
+		} else if (column === "mid") {
+			columnWidth = this.midColumnWidth;
+			columnDOM = this.midColumnDOM;
+		} else {
+			columnWidth = this.endColumnWidth;
+			columnDOM = this.endColumnDOM;
+		}
+
+		const currentlyHidden = columnWidth === "0px";
```

Comment:
The variable "currentlyHidden" will be evaluated to false, even if it's hidden, when the columnWidth is 0, but checked against "0px".
This will lead to different execution, and this code won't be even called:

```js
if (currentlyHidden) {
	columnDOM.style.width = columnWidth;

	// hide column with delay to allow the animation runs entirely
	columnDOM.addEventListener("transitionend", this.columnResizeHandler);
	return;
}
```